cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -49,28 +46,32 @@ public class InMemorySourceControlOperationRunner implements SourceControlOperat
   private static final Logger LOG = LoggerFactory.getLogger(InMemorySourceControlOperationRunner.class);
 
   @Inject
-  public InMemorySourceControlOperationRunner(RepositoryManagerFactory repoManagerFactory) {
+  InMemorySourceControlOperationRunner(RepositoryManagerFactory repoManagerFactory) {
     this.repoManagerFactory = repoManagerFactory;
   }
 
   @Override
   public PushAppResponse push(NamespaceId namespace, RepositoryConfig repoConfig,
                               ApplicationDetail appToPush, CommitMeta commitDetails)
     throws PushFailureException, NoChangesToPushException, AuthenticationConfigException {
-    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+    RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig);
+    try {
       try {
         repositoryManager.cloneRemote();
-      } catch (GitAPIException e) {
-        throw new PushFailureException("Failed to clone remote repository", e);
+      } catch (GitAPIException | IOException e) {
+        throw new PushFailureException(String.format("Failed to clone remote repository: %s", e.getMessage()));
       }
 
       LOG.info("Pushing application configs for : {}", appToPush);
 
       //TODO: CDAP-20371, Add retry logic here in case the head at remote moved while we are doing push
       return writeAppDetailAndPush(repositoryManager, appToPush, commitDetails);
-    } catch (IOException e) {
-      // This catch is because of RepositoryManager being AutoCloseable
-      throw new PushFailureException("Failed to initialize repository", e);
+    } finally {
+      try {
+        repositoryManager.close();
+      } catch (IOException e) {
+        LOG.warn("Failed to close on RepositoryManager: {}", e.getMessage());
```

Comment:
log messages are internal, used by system admins. They should normally log the full stacktrace otherwise the root cause is lost.

`LOG.warn("Failed to close the RepositoryManager, there may be leftover files", e)`

Reply:
Added back the root case logging